lib/src/client.dart

```diff
@@ -3213,13 +3232,15 @@ class Client extends MatrixApi {
   }
 
   final Map<String, DateTime> _keyQueryFailures = {};
+  Set<String>? _trackedUserIds;
 
   Future<void> updateUserDeviceKeys({Set<String>? additionalUsers}) async {
     try {
       final database = this.database;
       if (!isLogged()) return;
       final dbActions = <Future<dynamic> Function()>[];
-      final trackedUserIds = await _getUserIdsInEncryptedRooms();
+      final trackedUserIds =
+          _trackedUserIds ??= await _getUserIdsInEncryptedRooms();
```

Comment:
Is this potentially racy between a user signing out and this field being set? In theory the logout could be quicker, so the field could be set even after logout, which might lead to weird crypto issues?